pkg/parser.go

```
@@ -0,0 +1,189 @@
+	}
+
+	_, sectionExists := p.parsedData[sectionName]
+
+	if !sectionExists {
+		p.parsedData[sectionName] = make(map[string]string)
+	}
+
+	p.parsedData[sectionName][key] = value
+
+	return nil
+}
+
+// GetSectionNames returns a slice of section names.
+func (p *Parser) GetSectionNames() []string {
+	sectionNames := make([]string, 0, len(p.parsedData))
+
+	for sectionName := range p.parsedData {
+		sectionNames = append(sectionNames, sectionName)
+	}
+
+	return sectionNames
+}
+
+// GetSections returns a map of sections and their keys and values.
+func (p *Parser) GetSections() map[string]map[string]string {
+	return p.parsedData
+}
+
+// String returns a string representation of the parsed data.
+func (p *Parser) String() string {
+
+	var str string
+	for section, properties := range p.parsedData {
+		str += "[" + section + "]\n"
+		for key, value := range properties {
+			str += key + "=" + value + "\n"
+		}
+	}
+
+	str = strings.TrimSuffix(str, "\n")
+
+	return str
+}
+
+// SaveToFile saves the parsed data to a file.
+func (p *Parser) SaveToFile(filePath string) error {
+	err := os.WriteFile(filePath, []byte(p.String()), 0644)
+
+	if err != nil {
+		return ErrWritingToFile
+	}
+	return nil
+}
+
+func parseLines(lines []string) (map[string]map[string]string, error) {
+	parsedData := make(map[string]map[string]string)
+	var currentSection string
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+
+		if line == "" {
+			continue
+		}
+
+		if strings.HasPrefix(line, "[") && strings.HasSuffix(line, "]") {
+			sectionName := strings.TrimSpace(line[1 : len(line)-1])
+			if sectionName == "" {
+				return nil, ErrSectionIsEmpty
+			}
+			currentSection = sectionName
+			if _, exists := parsedData[currentSection]; !exists {
+				parsedData[currentSection] = make(map[string]string)
+			}
+			continue
+		}
+
+		if strings.Contains(line, "=") {
+			if strings.HasPrefix(line, "#") || strings.HasPrefix(line, ";") {
+				return nil, ErrCommentOnNewLine
+			}
```

why this should return an error